src/Sharpliner/AzureDevOps/ConditionedExpressions/ConditionBuilder.cs

```diff
@@ -105,10 +158,28 @@ internal ConditionBuilder(Conditioned<T>? parent = null)
     public Condition<T> Equal(string condition)
         => Link(new CustomCondition<T>(condition));
 
-    public Condition<T> Equal(string expression1, string expression2)
+    public Condition<T> Equal(StringRuntimeExpression expression1, StringRuntimeExpression expression2)
         => Link(new EqualityCondition<T>(true, expression1, expression2));
 
-    public Condition<T> NotEqual(string expression1, string expression2)
+    public Condition<T> Equal(IRuntimeExpression expression1, StringRuntimeExpression expression2)
+        => Link(new EqualityCondition<T>(true, expression1, expression2));
+
+    public Condition<T> Equal(StringRuntimeExpression expression1, IRuntimeExpression expression2)
+        => Link(new EqualityCondition<T>(true, expression1, expression2));
+
+    public Condition<T> Equal(IRuntimeExpression expression1, IRuntimeExpression expression2)
+        => Link(new EqualityCondition<T>(true, expression1, expression2));
+
+    public Condition<T> NotEqual(StringRuntimeExpression expression1, StringRuntimeExpression expression2)
+        => Link(new EqualityCondition<T>(false, expression1, expression2));
+
+    public Condition<T> NotEqual(IRuntimeExpression expression1, StringRuntimeExpression expression2)
+        => Link(new EqualityCondition<T>(false, expression1, expression2));
+
+    public Condition<T> NotEqual(StringRuntimeExpression expression1, IRuntimeExpression expression2)
+        => Link(new EqualityCondition<T>(false, expression1, expression2));
+
+    public Condition<T> NotEqual(IRuntimeExpression expression1, IRuntimeExpression expression2)
         => Link(new EqualityCondition<T>(false, expression1, expression2));
 
     public Condition<T> And(params Condition[] expressions)
```

Comment:
`And` and `Or` in my opinion should not support the new stuff, so I'd keep them as-is

Reply:
Done